### ISSUE TITLE
Improvement: Replace current chat system with a (web)socket system

### ISSUE THREAD

**Issue by DonMul** [FrankProjects/UltimateWarfare#82]
Probably have a chatserver running with react/ratchet. This way the chats won't be put through an AJAX system which polls from the server plus only a small backlog needs to be kept of the chat. This will reduce the load/strain on the hosting server

**Comment by frank9999** [comment]
Started working on this in #117, do you have experience implementing a secure authentication system using websockets? Not sure yet how to do this.

**Comment by DonMul** [comment]
There are 2 ways i've figured out so far to do websockets secure (WSS):
Using https://github.com/vakata/websocket as the lib to setup the server and then do something like this in your constructor of the server (and have it run in the background as a PHP script on a random server):
```
       $this->hostname = $hostname;
        $this->clients = [];

        $this->server = new \vakata\websocket\Server(
            'wss://0.0.0.0:' . $port,
            '/full/path/to/pem/file.pem'
        );

        $this->server->onConnect([$this, 'onConnect']);
        $this->server->onDisconnect([$this, 'onDisconnect']);
        $this->server->onMessage([$this, 'onMessage']);
```
Or make use of Ratchet ( http://socketo.me/ ) and so something like:
```

use Ratchet\Http\HttpServer;
use Ratchet\Server\IoServer;
use Ratchet\WebSocket\WsServer;
$httpSever =
                new HttpServer(
                    new WsServer(
                        new MessageComponentInterface( // This one needs to be replaced by your own server implementing this interface
                        )
                    )
                );

            $loop = \React\EventLoop\Factory::create();

            $secure_websockets = new \React\Socket\Server('0.0.0.0:' .  $port, $loop);
            $secure_websockets = new \React\Socket\SecureServer($secure_websockets, $loop, [
                'local_cert' => '/path/to/fullchain.pem',
                'local_pk' => '/path/to/privkey.pem',
                'verify_peer' => false
            ]);

            try {
                $secure_websockets_server = new \Ratchet\Server\IoServer($httpSever, $secure_websockets, $loop);
                $secure_websockets_server->run();
            } catch (\Throwable $throwable) {
            }
            ```

**Comment by frank9999** [comment]
Merged new websocket based ChatServer in #117
